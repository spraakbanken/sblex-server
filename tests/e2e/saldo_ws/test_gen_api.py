# import xml.etree.ElementTree as ET

import pytest
from fastapi import status
from httpx import AsyncClient

# EXPECTED_XML_RESPONSES = {
#     "empty": '<?xml version="1.0" encoding="UTF-8"?>\n<result>\n\n</result>',
#     "dväljas..vb.1": """<result>
# <gf>dväljas</gf>
#     <p>vb_vs_dväljas</p>
#     <ls>
#     <l>dväljas..1</l>
#     </ls>

# </result>
# """,
#     "dväljas..1": """<result>
# <lex>dväljas..1</lex>
#     <fm>bo..1</fm>
#     <fp>PRIM..1</fp>
#     <mfs></mfs>
#     <pfs></pfs>
#     <ls>
#     <l>dväljas..vb.1</l>
#     </ls>
# </result>
# """,
# }


class TestGenRoutes:
    @pytest.mark.parametrize(
        "paradigm, word, expected_response",
        [
            (
                "vb_vs_dväljas",
                "dväljas",
                [
                    {
                        "form": "dväljes",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "pres ind s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dväljs",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "pres ind s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dvaldes",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "pret ind s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dväljdes",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "pret ind s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dväljes",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "imper",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dväljs",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "imper",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dväljas",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "inf s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dvalts",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "sup s-form",
                        "p": "vb_vs_dväljas",
                    },
                    {
                        "form": "dvälts",
                        "gf": "dväljas",
                        "pos": "vb",
                        "inhs": [],
                        "msd": "sup s-form",
                        "p": "vb_vs_dväljas",
                    },
                ],
            ),
            #             ("d..nn.1", {}),
        ],
    )
    @pytest.mark.asyncio
    async def test_json_valid_input_returns_200(
        self,
        client: AsyncClient,
        paradigm: str,
        word: str,
        expected_response: list,
    ) -> None:
        res = await client.get(f"/gen/json/{paradigm}/{word}")

        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "application/json"
        assert res.json() == expected_response

    #     @pytest.mark.parametrize(
    #         "lid, expected_response_name",
    #         [
    #             (
    #                 "xxxxx..xx.1",
    #                 "empty",
    #             ),
    #             ("dväljas..vb.1", "dväljas..vb.1"),
    #             (
    #                 "xxxxx..1",
    #                 "empty",
    #             ),
    #             (
    #                 "dväljas..1",
    #                 "dväljas..1",
    #             ),
    #         ],
    #     )
    #     @pytest.mark.asyncio
    #     async def test_xml_valid_input_returns_200(
    #         self,
    #         client: AsyncClient,
    #         lid: str,
    #         expected_response_name: str,
    #     ) -> None:
    #         res = await client.get(f"/gen/xml/{lid}")
    #         assert res.status_code == status.HTTP_200_OK
    #         assert res.headers["content-type"] == "application/xml"

    #         expected_xml = ET.canonicalize(EXPECTED_XML_RESPONSES[expected_response_name])

    #         response_xml = ET.canonicalize(res.text)
    #         assert response_xml == expected_xml

    @pytest.mark.parametrize(
        "paradigm, word, expected_in_response",
        [
            ("vb_vs_dväljas", "dväljas", ["<td>vb_vs_dväljas</td>", "<td>dväljas</td>"]),
            ("xx_xx_xxx", "xxx", ["<p>paradigm ", " finns ej.</p>"]),
        ],
    )
    @pytest.mark.asyncio
    async def test_html_valid_lemma_returns_200(
        self,
        client: AsyncClient,
        paradigm: str,
        word: str,
        expected_in_response: list[str],
    ) -> None:
        res = await client.get(f"/gen/html/{paradigm}/{word}")

        assert res.status_code == status.HTTP_200_OK
        assert res.headers["content-type"] == "text/html; charset=utf-8"
        print(f"{res.text=}")
        for expect_in_response in expected_in_response:
            assert expect_in_response in res.text
