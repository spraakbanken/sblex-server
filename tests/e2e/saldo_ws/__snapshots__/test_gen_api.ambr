# serializer version: 1
# name: TestGenRoutes.test_html_valid_lemma_returns_200[vb_vs_dv\xe4ljas-dv\xe4ljas]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title>Böjningstabell | vb_vs_dväljas &#34;dväljas&#34;</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      
      <table border="1">
          <tr>
              <td><b>grundform</b></td>
              <td>dväljas</td>
          </tr>
          <tr>
              <td><b>mönster</b></td>
              <td>vb_vs_dväljas</td>
          </tr>
          <tr>
              <td><b>ordklass</b></td>
              <td>vb</td>
          </tr>
          
          <tr>
              <td colspan="2" style="text-align:center;"><b>böjningstabell</b></td>
  
          </tr>
          
          <tr>
              <td><i>pres ind s-form</i></td>
              <td>dväljs/dväljes</td>
          </tr>
          
          <tr>
              <td><i>pret ind s-form</i></td>
              <td>dväljdes/dvaldes</td>
          </tr>
          
          <tr>
              <td><i>imper</i></td>
              <td>dväljs/dväljes</td>
          </tr>
          
          <tr>
              <td><i>inf s-form</i></td>
              <td>dväljas</td>
          </tr>
          
          <tr>
              <td><i>sup s-form</i></td>
              <td>dvälts/dvalts</td>
          </tr>
          
      </table>
      
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestGenRoutes.test_html_valid_lemma_returns_200[xx_xx_xxx-xxx]
  '''
  <!DOCTYPE html>
  <html>
  
  <head>
    <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
    <link rel="shortcut icon" href="https://svn.spraakdata.gu.se/sb-arkiv/pub/images/saldo_icon.png">
    <link rel="stylesheet" type="text/css" href="http://testserver/static/saldo.css" />
    <title>Böjningstabell | xx_xx_xxx saknas</title>
    
    <script src="https://unpkg.com/htmx.org@1.9.8" crossorigin="anonymous"></script>
  </head>
  
  <body>
    <center>
      <p>
        <a href="http://testserver/fl/html"><img
            src="https://spraakbanken.gu.se/sites/spraakbanken.gu.se/files/img/saldo/saldo.gif" align="top"
            alt="SALDO: gå tillbaka till startsidan" /></a>
      </p>
    </center>
    <div id="output_table">
      
  <center>
      
      <p>paradigm 'xx_xx_xxx' finns ej.</p>
      
  </center>
  
    </div>
    <footer>
      <hr>
      <ul>
        <li><a href="https://spraakbanken.gu.se/saldo">SALDOs projektsida</a></li>
        <li><a href="https://svn.spraakdata.gu.se/repos/sblex/pub/saldo_instruktion.pdf">dokumentation (pdf)</a></li>
      </ul>
    </footer>
  </body>
  
  </html>
  '''
# ---
# name: TestGenRoutes.test_xml_valid_input_returns_200[vb_vs_dv\xe4ljas-dv\xe4ljas]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result><table><w>
              <form>dväljes</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>pres ind s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dväljs</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>pres ind s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dvaldes</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>pret ind s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dväljdes</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>pret ind s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dväljes</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>imper</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dväljs</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>imper</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dväljas</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>inf s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dvalts</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>sup s-form</msd>
              <p>vb_vs_dväljas</p>
          </w><w>
              <form>dvälts</form>
              <gf>dväljas</gf>
              <pos>vb</pos>
              <inhs></inhs>
              <msd>sup s-form</msd>
              <p>vb_vs_dväljas</p>
          </w></table></result>
  '''
# ---
# name: TestGenRoutes.test_xml_valid_input_returns_200[xx_xx_xxx-xxx]
  '''
  <?xml version="1.0" encoding="UTF-8"?>
  <result></result>
  '''
# ---
